mewarpx/examples/thermionic_diode_rz_cylinder.py

```diff
@@ -0,0 +1,194 @@
+            use_Schottky=self.USE_SCHOTTKY
+        )
+
+        #######################################################################
+        # Clean up and output RunInfo                                         #
+        #######################################################################
+
+        runvars = VacuumTEC_Cylinder.__dict__.copy()
+        runvars.update(self.__dict__)
+
+        self.runinfo = runinfo.RunInfo(
+            local_vars=runvars,
+            electrode_params={"CATHODE_A": self.CATHODE_A,
+                              "CATHODE_TEMP": self.CATHODE_TEMP,
+                              "CATHODE_PHI": self.CATHODE_PHI,
+                              "ANODE_PHI": self.ANODE_PHI},
+            surface_dict={'cathode': self.cathode, 'anode': self.anode},
+            injector_dict={'cathode': self.cathode_injector},
+        )
+        # overwrite runinfo area to properly normalize currents
+        self.runinfo.area = self.cathode_emitter.area
+
+        #######################################################################
+        # Add diagnostics                                                     #
+        #######################################################################
+
+        self.text_diag = diags.TextDiag(
+            diag_steps=self.DIAG_STEPS, preset_string='perfdebug'
+        )
+        # self.field_diag = diags.FieldDiagnostic(
+        #    diag_steps=self.DIAG_STEPS, style='roelof', save_pdf=False
+        # )
+        self.fluxdiag = diags.FluxDiagnostic(
+            diag_steps=self.DIAG_STEPS,
+            runinfo=self.runinfo,
+            check_charge_conservation=False,
+            overwrite=True
+        )
+
+        #######################################################################
+        # Initialize run and print diagnostic info                            #
+        #######################################################################
+
+        mwxrun.init_run()
+
+
+    def run_sim(self):
+
+        mwxrun.simulation.step(self.TOTAL_TIMESTEPS)
+
+        if self.SAVE and mwxrun.me == 0:
+            key = ('scrape', 'anode', 'electrons')
+            J_diode = self.fluxdiag.ts_dict[key].get_averagevalue_by_key('J')
+            print(f'{self.V_ANODE_CATHODE}    {J_diode}')
+            with open(f'results_d_{int(self.D_CA*1e6)}.dat', 'a') as f:
+                f.write(f'{self.V_ANODE_CATHODE}    {J_diode}\n')
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--V', help='bias voltage in Volt', type=float, default=0)
+parser.add_argument('--steps', help='set the number of simulation steps',
+                    type=int, default=3000)
+parser.add_argument('--save', help='save voltage and current pairs',
+                    default=False, action='store_true')
+args, left = parser.parse_known_args()
+sys.argv = sys.argv[:1]+left
+
+run = VacuumTEC_Cylinder()
+run.V_ANODE_CATHODE = args.V
+run.TOTAL_TIMESTEPS = args.steps
+run.SAVE = args.save
```

Comment:
```suggestion
```
Passing these as init arguments is a much cleaner separation between class interface and outside code.